### ISSUE TITLE
[github-actions] improve build-actions

### ISSUE THREAD

**Issue by hgy59** [SynoCommunity/spksrc#4018]
We should improve the following for gh build-actions
-  ~Add variable to Makefiles for better recognition of noarch packages (e.g. use `NO_ARCH = 1` instead of `ARCH =` in spk Makefiles (and adjust the /mk/*.mk files for this)~
no need for this. It is easy to parse for `override ARCH`
- [x] Split build for `noarch` and arch specific packages (to avoid noarch build for each arch, what could also lead to conflicts in parallel builds)
- [x] Find a solution to detect and report real build errors (exclude errors due to `UNSUPPORTED_ARCH`). May be we need an extra task to analyze the build log.
- [ ] Find a solution to report download errors.
- [x] Optimize the build of `ffmpeg` dependent packages with caching (@th0ma7 or is this already done?)
- [ ] Add separate build for packages with `REQ_KERNEL` to use a different arch-matrix (avoid generic archs)
- [ ] Add a cache for downloaded sources (distrib folder)

**Comment by publicarray** [comment]
While all of these features are nice to have. I don't want to take too much time out of synocommunity to maintaining the automation for the project. The automation should remain simple and do it's job: https://xkcd.com/1319/
![automation](https://user-images.githubusercontent.com/5497998/84089901-96e70d00-aa33-11ea-9f4b-cb9704936456.png)

**Comment by publicarray** [comment]
Just ignore my previous comment. I think these are good goals to set 👍

**Comment by th0ma7** [comment]
@hgy59 I did much improved the `tvheadend` package building against `ffmpeg` build part of PR #4005.  It allows a much faster build time (around 5 mins per arch instead of 30-60 mins).

Still, I want to point your attention on this: in order to work a build of `ffmpeg` must be residing permanently so dependant packages builds can be fast-tracked.

For instance I just updated PR #4005 and it's currently running big-time to do a full rebuild thus monopolizing CPU big time which in turns may lead to using all our processing time.

`ffmpeg` is sooooo huge that it may need to be managed differently (or not at all) along with changes to its dependencies and dependant package of it.

**Comment by hgy59** [comment]
Instead of reporing real build errors, I just added a `Build Status` job. The Build writes each failed package into a separate file and the status job fails when this file contains entries.

**Comment by th0ma7** [comment]
Hi @hgy59 I received a flood of emails from github-actions this morning...  
![github-action-gmail](https://user-images.githubusercontent.com/17141866/84884811-302fa800-b060-11ea-858d-3430073fae83.png)

I just noticed that it actually "actionned" on my cloned repository as well for each commit.

So a few things comes to mind as for projects as large as `ffmpeg` this will waste tremendous amount of cpu cycles.  
a) should not action by default on local clones.  This really should be user based activated/triggered.
b) Ability to trigger on demand would help for both local clones & specific projects (again, no action by default + manually triggered option).
c) A "queue-wait" timer such as if you push a commit it doesn't build right-away: instead if waits a bit (e.g. 15-40-60 mins?) to see if this job isn't being superseeded by another one through a new commit, if so drop current pending job and leave the floor next instead.

**Comment by hgy59** [comment]
@th0ma7 see this article on how to disable actions for your local repository:
https://help.github.com/en/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository

**Comment by th0ma7** [comment]
> @th0ma7 see this article on how to disable actions for your local repository:
> https://help.github.com/en/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository

Thnx!  much appreciated @hgy59 .  I'll have a read and fix that on my end.

**Comment by hgy59** [comment]
Even not all of the initial positions are solved I close this issue, as I will not work on it in the near future.
